### PR TITLE
feat: reorganize documentation

### PR DESCRIPTION
**Pull request by SdgJlbl**
The goal of this PR is to reorganise the documentation according to the [Diátaxis framework](https://diataxis.fr/) to facilitate access. 

**TODO**

- [ ] fix Binder (missing assets dir + not installed dependencies)
- [x] check all links are still valid in examples

### CODE REVIEW

**Comment by guilhem-barthes** [comment]
🙏

**Comment by SdgJlbl** [comment]
@ThibaultFy Is it expected that Binder is running all examples (and rebuilding the whole documentation) at launch time? It does not seem very scalable in the long term 🤔

File: docs/source/how-to/deploying-substra/index.rst
Comment (oleobal) [inline docs/source/how-to/deploying-substra/index.rst:2]: It seems to me this section isn't just how-to guides, are you planning on splitting it up further?

Comment (SdgJlbl) [reply docs/source/how-to/deploying-substra/index.rst:2]: It was a first pass on trying to organise stuff, but maybe the walk-through should be moved to the tutorials indeed. 
Open to suggestion as to what we want to put where :)

Comment (oleobal) [reply docs/source/how-to/deploying-substra/index.rst:2]: Since we're serving the needs of different people, perhaps we could add an additional layer on top of the diataxis one?

So you'd select **Persona** (DS, developer, SRE) and then **mode** (reference, tutorial, how-to, explanation)

Apologies, this is a pretty fundamental suggestion to make when you've already made all these changes

Comment (SdgJlbl) [reply docs/source/how-to/deploying-substra/index.rst:2]: Personally I'd rather have the **persona** under the **mode** rather than the other way. The API reference is going to be the same for all personas, and some other things may be hard to classify (some part of the explanations might be of interest to everyone, whereas other parts will interest advanced users, be they data scientists or SRE). 
What do you think?

Comment (oleobal) [reply docs/source/how-to/deploying-substra/index.rst:2]: It's a good point that there is a lot of common from one to the other. Although we can possibly use links (or even straight copy, since we have an automated build system) to mitigate this.

I'm no expert (let's hire a technical writer!) so my opinion is just that, but I feel persona is more fundamental than mode.

In a single work session there's a good chance I'll access multiple modes, but I'll probably stick to the same persona (true for my use case, but maybe as an MLE you need to access both DS and developer personas?).

And I'd rather have a big switch at the top of the screen (a bit radical lol, but that's how openshift does it) than "if you are deploying substra" plastered the middle of the text

Comment (SdgJlbl) [reply docs/source/how-to/deploying-substra/index.rst:2]: Thinking about it, I'm not sure that I would consider the deployment walk-through as a tutorial, in the sense that tutorials are supposed to be focused on teaching new things. 
In this lens, the walk-through is more of a long how-to guide for deploying. 
I've arranged / renamed things a bit, let me know what you think.

File: examples/substrafl/go_further/run_diabetes_substrafl.py
Comment (ThibaultFy) [inline examples/substrafl/go_further/run_diabetes_substrafl.py:502]: Woops.. good catch

**Review by ThibaultFy — COMMENTED** [review]
Love what you did 😍  thanks a lot for this OS contrib !! 
I would love to approve but we should wait for Romain review IMO.

@RomainGoussault you should take a look at this !

File: docs/source/how-to/developing-substra/index.rst
Comment (oleobal) [inline docs/source/how-to/developing-substra/index.rst:2]: The titles for these sections are

- How-to guides for Substra developers
- How-to guides for deploying Substra
- How-to guides for Substra users

Do you think we could standardize on one formulation? such as "for developing on Substra" and "for using Substra". (unfortunately "deployers" isn't a word.. But I think it might be better to focus on action than on title)

**Review by oleobal — APPROVED** [review]
how-to guide titles are not all standardized but IMO let's merge it as-is and address this in a separate PR

**Pull request merged**
